packages/controllers/src/permissions/rpc-methods/requestPermissions.ts

```diff
@@ -0,0 +1,63 @@
+import type { PermittedHandlerExport } from '@mm-snap/rpc-methods/types';
+import type {
+  JsonRpcEngineEndCallback,
+  JsonRpcRequest,
+  PendingJsonRpcResponse,
+} from 'json-rpc-engine';
+import { nanoid } from 'nanoid';
+import { MethodNames } from '../enums';
+
+import { invalidParams } from '../errors';
+import type { Permission, RequestedPermissions } from '../Permission';
+
+export const requestPermissionsHandler: PermittedHandlerExport<
```

Comment:
What is a `PermittedHandlerExport`?

Reply:
Basically I'm having trouble following how the entire `rpc-methods` directory is used, and what these middleware properties are used for. Where is this description used? What is a "hook"?